index.js

```diff
@@ -41,13 +41,17 @@ module.exports = function (cb) {
 			isReachable(hostnames, cb);
 		}
 
-		udpSocket.unref();
+		udpSocket.close();
+		udpSocket = null;
 	});
 
 	udpSocket.send(payload, 0, payload.length, 53, server, function () {
 		setTimeout(function () {
 			// We ran into the timeout, we're offline with high confidence
 			cb(null, false);
+			if (udpSocket) {
+				udpSocket.close();
+			}
 		}, timeout);
 
 		udpSocket.unref();
```

Comment:
This `unref` should enable node to exit if all it's doing is checking if the machine is online.  I guess somebody wanted this feature at some point?   But I'm not attached to the `unref` and will remove it if we don't want to support that feature.